x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexIT.java

```diff
@@ -81,6 +83,18 @@ public void testFollowIndex() throws Exception {
         }
     }
 
+    public void testFollowNonExistingLeaderIndex() throws Exception {
```

Comment:
I would have preferred an unit test here, but the behavior between following a local / remote index is different and the `CcrLicenseChecker` fetch the leader index metadata and then immediately fetch history uuids, which caused a NPE if leader index metadata is null.